### ISSUE TITLE
Implement Batch Normalisation Layer

### ISSUE THREAD

**Issue by Manthan-R-Sheth** [mlpack/mlpack#1273]
I would like to begin the task of implementing deep learning modules with the implementation of [Batch Normalisation Layer](https://arxiv.org/abs/1502.03167). This is due to the wide advantage of using BatchNorm for DL tasks. It can be used for FFN as well as Convolution Networks.

@zoq @lozhnikov @rcurtin, shall i go ahead with this task?

**Comment by zoq** [comment]
BatchNorm would be a great to have, there is an open PR https://github.com/mlpack/mlpack/pull/955 which I think is a good start, so if you like, clone the code and resolve the remaining issues.

**Comment by sourabhvarshney111** [comment]
@Manthan-R-Sheth I think we have completed our work for this. Can we close it now? What say?

**Comment by zoq** [comment]
Right, let's close this one.
